account-kit/java/signer-sdk/src/main/java/com/alchemy/aa/Stamper.java

```diff
@@ -1,140 +1,65 @@
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.ObjectWriter;
-import com.google.common.primitives.Bytes;
-import com.google.crypto.tink.BinaryKeysetReader;
-import com.google.crypto.tink.CleartextKeysetHandle;
-import com.google.crypto.tink.KeysetHandle;
-import com.google.crypto.tink.hybrid.HpkePrivateKey;
 import com.google.crypto.tink.subtle.Base64;
-import com.google.crypto.tink.subtle.EllipticCurves;
-
 import com.google.crypto.tink.subtle.Hex;
-import com.google.protobuf.InvalidProtocolBufferException;
-import java.io.IOException;
-import java.nio.ByteBuffer;
 import java.security.GeneralSecurityException;
 import java.security.KeyFactory;
 import java.security.Security;
-import java.security.interfaces.ECPrivateKey;
-import java.security.interfaces.ECPublicKey;
 import java.security.Signature;
-import java.security.spec.ECPoint;
-import javax.annotation.Nonnull;
-import org.bitcoinj.base.Base58;
-import org.bouncycastle.jce.ECNamedCurveTable;
+import java.security.interfaces.ECPrivateKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import lombok.Getter;
+import lombok.Setter;
 import org.bouncycastle.jce.provider.BouncyCastleProvider;
-import org.bouncycastle.jce.spec.ECPublicKeySpec;
 
 public class Stamper {
 
     public record Stamp(String stampHeaderName, String stampHeaderValue) {
     }
 
     public record APIStamp(String publicKey, String scheme, String signature) {
-    }
-
-    public static String toJson(APIStamp apiStamp) throws JsonProcessingException {
-        ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
-        return ow.writeValueAsString(apiStamp);
-    }
 
-    public Stamper(TEKManager tekManager) {
-        this();
-
-        this.tekManager = tekManager;
+        public String toJson() throws JsonProcessingException {
+            ObjectWriter ow = new ObjectMapper().writer().withDefaultPrettyPrinter();
+            return ow.writeValueAsString(this);
+        }
     }
 
-    /**
-     * @param targetPrivateKeyHex
-     *            Base64 private key
-     */
-    public Stamper(@Nonnull String targetPrivateKeyHex) throws GeneralSecurityException, IOException {
-        this();
-        byte[] keysetBytes = Base64.decode(targetPrivateKeyHex);
-        KeysetHandle privateKeysetHandle = CleartextKeysetHandle.read(BinaryKeysetReader.withBytes(keysetBytes));
 
-        this.tekManager = TEKManager.InitializeTEKManagerFromKeySetHandle(privateKeysetHandle);
-    }
 
-    public Stamper(@Nonnull byte[] targetPrivateKey) throws GeneralSecurityException, IOException {
-        this();
-        KeysetHandle privateKeysetHandle = CleartextKeysetHandle.read(BinaryKeysetReader.withBytes(targetPrivateKey));
-        this.tekManager = TEKManager.InitializeTEKManagerFromKeySetHandle(privateKeysetHandle);
-    }
+    @Getter
+    @Setter
+    private final CredentialBundle credentialBundle;
```

Comment:
It doesn't make sense to have a setter on a `final` variable. Remove either the setter or `final`.

Reply:
I meant removing the setter and making it final cause it doesn't make much sense to be able to change it